nemo/lightning/pytorch/accelerate/transformer_engine.py

```diff
@@ -0,0 +1,90 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from types import MethodType
+
+import torch
+from transformer_engine import pytorch as te
+
+
+def accelerate(model, fp8_autocast=False):
+    _accelerate(model)
+    if fp8_autocast:
+        apply_fp8_autocast(model)
+
+
+def _accelerate(model):
+    for name, module in model.named_modules():
+        if isinstance(module, torch.nn.Linear):
+            has_bias = module.bias is not None
+            if any(p % 16 != 0 for p in module.weight.shape):
+                continue
+            te_module = te.Linear(
+                module.in_features, module.out_features, bias=has_bias, params_dtype=module.weight.dtype
+            )
+            with torch.no_grad():
```

Comment:
I see there's torch.no_grad here but not in layernorm/rmsnorm, I wonder if it would be possible to move the torch.no_grad as a function decorator, i.e.:
```
@torch.no_grad
def _accelerate(model):
....
```

Reply:
Done.